src/bout++.cxx

```diff
@@ -189,60 +183,96 @@ int BoutInitialise(int &argc, char **&argv) {
                 "physics model source (e.g. %s.cxx)\n"),
               argv[0]);
 
-      return -1;
+      std::exit(EXIT_SUCCESS);
     }
   }
+
   bool color_output = false; // Will be set true if -c is in the options
+  std::vector<std::string> original_argv;
+
   for (int i=1;i<argc;i++) {
     if (string(argv[i]) == "-d") {
       // Set data directory
       if (i+1 >= argc) {
         fprintf(stderr, _("Usage is %s -d <data directory>\n"), argv[0]);
         return 1;
       }
-      i++;
-      data_dir = argv[i];
-      
+
+      data_dir = argv[++i];
+
+      original_argv.push_back(argv[i - 1]);
+      original_argv.push_back(argv[i]);
```

Comment:
Why duplicate the push_back code, and not take a full copy initially?
Also, in case this is not an option we are processing here, it does not get copied, as far as I can see.
Is this intentional?